js/src/logger.ts

```diff
@@ -3291,7 +3294,7 @@ function startSpanAndIsLogger<IsAsyncFlush extends boolean = false>(
         components.data.object_type === SpanObjectTypeV3.PROJECT_LOGS &&
         // Since there's no parent logger here, we're free to choose the async flush
         // behavior, and therefore propagate along whatever we get from the arguments
-        !args?.asyncFlush,
+        (args?.asyncFlush === undefined ? false : !args?.asyncFlush),
```

Comment:
A simpler boolean expression would be `args?.asyncFlush === false`.

Reply:
Done